### PR TITLE
Swap out Optim.jl NelderMead.jl

### PR DESCRIPTION
**Pull request by jwscook**
This PR removes Optim.jl as a dependency and replaces it with [NelderMead.jl](https://github.com/jwscook/NelderMead.jl) given the context of https://github.com/jwscook/NelderMead.jl/issues/7.

A point worth noting is that the size of the simplex is taken to be the standard deviation of the starting position (or 1 in the case that the standard deviation is zero). There may be a better simplex size given the range of sensible values of $\alpha_0$; please advise.

Closes https://github.com/jwscook/NelderMead.jl/issues/7.

### CODE REVIEW

File: src/TableTransforms.jl [hunk 1]
```diff
@@ -16,7 +16,7 @@ using PrettyTables
 using AbstractTrees
 using CategoricalArrays
 using Random
-using Optim: optimize, minimizer
+using NelderMead
```
Comment (juliohm) [inline src/TableTransforms.jl:19]:
```suggestion
using NelderMead: optimise
```

File: src/transforms/projectionpursuit.jl [hunk 2]
```diff
@@ -111,8 +111,10 @@ end
 
 function neldermead(transform, Z, α₀)
   f(α) = -pindex(transform, Z, α ./ norm(α))
-  op = optimize(f, α₀)
-  minimizer(op)
+  l = std(α₀)
+  (minimiser, _) = NelderMead.optimise(f, α₀, iszero(l) ? one(l) : l,
+                                       xtol_rel=10eps())
+  minimiser
```
Comment (juliohm) [inline src/transforms/projectionpursuit.jl:117]:
```suggestion
  res = optimise(f, α₀, iszero(l) ? one(l) : l, xtol_rel=10eps())
  first(res)
```

File: test/Project.toml [hunk 3]
```diff
@@ -11,6 +11,7 @@ ReferenceTests = "324d217c-45ce-50fc-942e-d289b448e8cf"
 ScientificTypes = "321657f4-b219-11e9-178b-2701a2544e81"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
+TableTransforms = "0d432bfd-3ee1-4ac1-886a-39f05cc69a3e"
```
Comment (juliohm) [inline test/Project.toml:14]:
```suggestion
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @jwscook for the PR ❤️ Regarding the initial guess, it is guaranteed to have norm 1. Do you have suggestions for the simplex size in this case?

File: test/Project.toml (hunk 3 above)
Comment (jwscook) [reply test/Project.toml:14]: Woops! Good catch.

**Comment by jwscook** [comment]
> Thank you @jwscook for the PR ❤️ Regarding the initial guess, it is guaranteed to have norm 1. Do you have suggestions for the simplex size in this case?

I'm not entirely sure what the right size should be, but after some trial and error a value of 1/2 seems to work.

File: src/transforms/projectionpursuit.jl [hunk 4]
```diff
@@ -111,8 +111,8 @@ end
 
 function neldermead(transform, Z, α₀)
   f(α) = -pindex(transform, Z, α ./ norm(α))
-  op = optimize(f, α₀)
-  minimizer(op)
+  res = optimise(f, α₀, 1/2, xtol_rel=10eps())
+  first(res)
```
Comment (juliohm) [inline src/transforms/projectionpursuit.jl:115]:
```suggestion
  optimise(f, α₀, 1/2, xtol_rel=10eps()) |> first
```

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
That is lovely @jwscook. I will just wait for the tests to pass once again and will then merge it. Did you consider moving NelderMead.jl to JuliaOptim or some other organization with multiple maintainers? It should get more visibility there. :)

**Pull request merged**